packages/jest-snapshot/src/__tests__/utils.test.js

```diff
@@ -195,12 +195,74 @@ test('serialize handles \\r\\n', () => {
 
 describe('DeepMerge', () => {
   it('Correctly merges objects with property matchers', () => {
-    const target = {data: {bar: 'bar', foo: 'foo'}};
-    const matcher = expect.any(String);
-    const propertyMatchers = {data: {foo: matcher}};
+    /* eslint-disable sort-keys */
+    // to keep keys in numerical order rather than alphabetical
+    const target = {
+      data: {
+        one: 'one',
+        two: 'two',
+        three: [
+          {
+            four: 'four',
+            five: 'five',
+          },
+          // Include an array element not present in the propertyMatchers
+          {
+            six: 'six',
+            seven: 'seven',
+          },
+        ],
+      },
+    };
+    // Don't use `expect.any(string)` since that will cause a false positive
```

Comment:
Alternatively perhaps use Node's deep equal to "escape" asymmetric matchers during the equality checks? Still weird enough to deserve a comment, but gets this test closer to the real use case again

Reply:
Can you elaborate on `assert.deepEqual()` being used to `escape`? Sorry - I'm not following. I do understand the need to get this test closer to the real use case though.

Reply:
I mean importing `deepStrictEqual` from `assert` and using it instead of `expect().toStrictEqual()` in this exceptional case, with a comment that it's  being done so that `expect.any()` is being treated as a normal value, because `expect` would handle it specially. That should work, right?

Reply:
I understand now - thanks! I have updated the test implementation and it looks like it's working properly.